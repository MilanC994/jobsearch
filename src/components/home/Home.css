.home-search-job {
  padding: 170px 56px 205px;
}
.home-search-job input[type='text'] {
  padding: 30px 10px;
  line-height: 28px;
  font-size: 25px;
  width: 43%;
  border-radius: 0%;
}
.home-search-job button {
  font-size: 20px;
  padding: 14px 35px;
  border-radius: 0%;
}
.home-search-job h1 {
  font-size: 50px;
  padding-bottom: 14px;
  border-radius: 0%;
}
.home-search-job p {
  font: normal 16px/24px Inter, Helvetica, Arial, sans-serif;
  color: #393f47;
}
.city-card > img {
  height: 300px;
}
@media (max-width: 900px) {
  .home-search-job .form-inline {
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
    -ms-flex-direction: row;
    flex-direction: row;
    margin-top: 40px;
    display: block !important;
  }
  .home-search-job input[type='text'],
  button {
    width: 100%;
  }
}
.portfolio > h2 {
  color: #0b0d0e;
  font: 700 28px/40px Inter, Helvetica, Arial, sans-serif;
}
.portfolio > p {
  font: normal 18px/28px Inter, Helvetica, Arial, sans-serif;
}
@media (min-width: 1200px) and (max-width: 1550px) {
  .card-column {
    flex: 0 0 50% !important;
    max-width: 50% !important;
  }
}
