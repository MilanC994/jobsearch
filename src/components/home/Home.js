import React, { useCallback } from 'react'
import Navbar from '../navbar/Navbar'
import CityCard from './CityCard'
import './Home.css'
import useInputField from '../../hooks/useInputField'
import { setParams } from '../../redux/actions'
import { useDispatch } from 'react-redux'
import { useHistory } from 'react-router-dom'
function Home() {
  const cityLinks = [
    {
      imgLink: 'https://jooble.org/images/cities/us/33025.jpg',
      cityName: 'New York, NY',
      numOfVacancies: 65.535,
    },
    {
      imgLink: 'https://jooble.org/images/cities/us/1.jpg',
      cityName: 'Washington DC',
      numOfVacancies: 56.085,
    },
    {
      imgLink: 'https://jooble.org/images/cities/us/11886.jpg',
      cityName: 'Chicago, IL',
      numOfVacancies: 48.082,
    },
    {
      imgLink: 'https://jooble.org/images/cities/us/10460.jpg',
      cityName: 'Atlanta, GA',
      numOfVacancies: 38.497,
    },
  ]
  const dispatch = useDispatch()
  const history = useHistory()

  const { input: keywordsField } = useInputField('keywords', '', () => true)
  const { input: locationField } = useInputField('location', '', () => true)
  const setParameters = useCallback(() => {
    dispatch(
      setParams({
        keywords: keywordsField.value,
        location: locationField.value,
        page: 1,
      })
    )
    history.push('/jobs')
  }, [dispatch, keywordsField.value, locationField.value, history])

  return (
    <div className="container-fluid">
      <Navbar home={true} />
      <div className="home-search-job container">
        <h1>Your new job, vacancies all over the USA</h1>
        <p>
          More than 583800 current vacancies from 21470 sites available to you.
          Find your new job today.
        </p>
        <form className="form-inline">
          <input
            placeholder="What position are you looking for"
            type="text"
            className="form-control"
            {...keywordsField}
          />
          <input
            placeholder="Location"
            {...locationField}
            type="text"
            className="form-control"
          />
          <button
            type="button"
            onClick={setParameters}
            className="btn btn-primary"
          >
            Find Jobs
          </button>
        </form>
      </div>
      <hr />
      <div className="portfolio container-fluid">
        <h2>Job search by city, job in USA</h2>
        <p>
          More than 60 regions USA available for search. Find a job that matches
          your request.
        </p>
        <div className="row">
          {cityLinks.map(city => {
            return (
              <div
                key={city.imgLink}
                className="card-column col-xl-3 col-lg-6 col-md-12"
              >
                <CityCard city={city} />
              </div>
            )
          })}
        </div>
      </div>
    </div>
  )
}

export default Home
