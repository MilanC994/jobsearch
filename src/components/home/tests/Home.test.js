import React from 'react'
import Home from '../Home'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { mountToJson } from 'enzyme-to-json'
import { StaticRouter as Router } from 'react-router-dom'

const mockHistoryPush = jest.fn()

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}))

const createStore = configureMockStore([thunk])

const setUp = initialState => {
  const store = createStore(initialState)

  const wrapper = mount(
    <Provider store={store}>
      <Router>
        <Home />
      </Router>
    </Provider>
  ).childAt(0)
  return wrapper
}

describe('Header test', () => {
  const initialState = {
    user: {
      user: {},
      loggedIn: false,
    },
    jobs: {
      params: {
        keywords: '',
        location: '',
      },
    },
  }

  test('Renders correctly', () => {
    const wrapper = setUp(initialState)
    expect(mountToJson(wrapper)).toMatchSnapshot()
  })
  test('Redirects to jobs route on button push', () => {
    const wrapper = setUp(initialState)
    const element = wrapper.find('#find-jobs-button')
    expect(element.exists()).toBeTruthy()
    element.simulate('click')
    expect(mockHistoryPush).toHaveBeenCalledWith('/jobs')
  })
})
