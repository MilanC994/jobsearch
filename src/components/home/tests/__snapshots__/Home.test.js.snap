// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test Renders correctly 1`] = `
<StaticRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Home>
      <div
        className="container-fluid"
      >
        <Navbar
          home={true}
        >
          <nav
            className="bootstrap-navbar navbar navbar-expand-lg navbar-light"
          >
            <span
              className="navbar-brand"
            >
              <Link
                id="app-logo-link"
                to="/"
              >
                <LinkAnchor
                  href="/"
                  id="app-logo-link"
                  navigate={[Function]}
                >
                  <a
                    href="/"
                    id="app-logo-link"
                    onClick={[Function]}
                  >
                    <img
                      alt="logo "
                      className="align-top logo"
                      height="60"
                      src="jooble_logo.png"
                      width="150"
                    />
                  </a>
                </LinkAnchor>
              </Link>
            </span>
            <div
              className="fixed-div"
            />
            <div
              className="align-top navbar-links justify-content-end navbar-nav"
            >
              <form
                className="form-inline"
                id="login-register-form"
              >
                <div
                  className="mr-2 btn-group"
                  role="group"
                >
                  <Link
                    id="register-link"
                    to="/register"
                  >
                    <LinkAnchor
                      href="/register"
                      id="register-link"
                      navigate={[Function]}
                    >
                      <a
                        href="/register"
                        id="register-link"
                        onClick={[Function]}
                      >
                        <button
                          className="mx-2 btn btn-outline-primary btn-md"
                          id="register"
                        >
                          Register
                        </button>
                      </a>
                    </LinkAnchor>
                  </Link>
                  <Link
                    id="login-link"
                    to="/login"
                  >
                    <LinkAnchor
                      href="/login"
                      id="login-link"
                      navigate={[Function]}
                    >
                      <a
                        href="/login"
                        id="login-link"
                        onClick={[Function]}
                      >
                        <button
                          className="mx-2 btn btn-outline-secondary btn-md"
                          type="button"
                        >
                          Login
                        </button>
                      </a>
                    </LinkAnchor>
                  </Link>
                </div>
              </form>
            </div>
          </nav>
        </Navbar>
        <div
          className="home-search-job container"
        >
          <h1>
            Your new job, vacancies all over the USA
          </h1>
          <p
            className="mt-3"
          >
            More than 583800 current vacancies from 21470 sites available to you. Find your new job today.
          </p>
          <form
            className="form-inline"
          >
            <input
              className="form-control"
              error={null}
              name="keywords"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="What position are you looking for"
              type="text"
              value=""
            />
            <input
              className="form-control"
              error={null}
              name="location"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Location"
              type="text"
              value=""
            />
            <button
              className="btn btn-primary"
              id="find-jobs-button"
              onClick={[Function]}
              type="button"
            >
              Find Jobs
            </button>
          </form>
        </div>
        <hr />
        <div
          className="portfolio container-fluid"
        >
          <h2
            className="mb-3"
          >
            Job search by city, job in USA
          </h2>
          <p
            className="mb-5"
          >
            More than 60 regions USA available for search. Find a job that matches your request.
          </p>
          <div
            className="row"
          >
            <div
              className="card-column col-xl-3 col-lg-6 col-md-12"
              key="https://jooble.org/images/cities/us/33025.jpg"
            >
              <CityCard
                city={
                  Object {
                    "cityName": "New York, NY",
                    "imgLink": "https://jooble.org/images/cities/us/33025.jpg",
                    "numOfVacancies": 65.535,
                  }
                }
              >
                <div
                  className="bg-dark text-white city-card card mb-3"
                >
                  <img
                    alt="Card poster"
                    className="card-img"
                    src="https://jooble.org/images/cities/us/33025.jpg"
                  />
                  <div
                    className="card-img-overlay"
                  >
                    <div
                      className="container-fluid"
                      style={
                        Object {
                          "bottom": " 0px",
                          "position": "absolute",
                        }
                      }
                    >
                      <div
                        className="card-title h5"
                      >
                        New York, NY
                      </div>
                      <p
                        className="card-text"
                      >
                        65.535
                      </p>
                    </div>
                  </div>
                </div>
              </CityCard>
            </div>
            <div
              className="card-column col-xl-3 col-lg-6 col-md-12"
              key="https://jooble.org/images/cities/us/1.jpg"
            >
              <CityCard
                city={
                  Object {
                    "cityName": "Washington DC",
                    "imgLink": "https://jooble.org/images/cities/us/1.jpg",
                    "numOfVacancies": 56.085,
                  }
                }
              >
                <div
                  className="bg-dark text-white city-card card mb-3"
                >
                  <img
                    alt="Card poster"
                    className="card-img"
                    src="https://jooble.org/images/cities/us/1.jpg"
                  />
                  <div
                    className="card-img-overlay"
                  >
                    <div
                      className="container-fluid"
                      style={
                        Object {
                          "bottom": " 0px",
                          "position": "absolute",
                        }
                      }
                    >
                      <div
                        className="card-title h5"
                      >
                        Washington DC
                      </div>
                      <p
                        className="card-text"
                      >
                        56.085
                      </p>
                    </div>
                  </div>
                </div>
              </CityCard>
            </div>
            <div
              className="card-column col-xl-3 col-lg-6 col-md-12"
              key="https://jooble.org/images/cities/us/11886.jpg"
            >
              <CityCard
                city={
                  Object {
                    "cityName": "Chicago, IL",
                    "imgLink": "https://jooble.org/images/cities/us/11886.jpg",
                    "numOfVacancies": 48.082,
                  }
                }
              >
                <div
                  className="bg-dark text-white city-card card mb-3"
                >
                  <img
                    alt="Card poster"
                    className="card-img"
                    src="https://jooble.org/images/cities/us/11886.jpg"
                  />
                  <div
                    className="card-img-overlay"
                  >
                    <div
                      className="container-fluid"
                      style={
                        Object {
                          "bottom": " 0px",
                          "position": "absolute",
                        }
                      }
                    >
                      <div
                        className="card-title h5"
                      >
                        Chicago, IL
                      </div>
                      <p
                        className="card-text"
                      >
                        48.082
                      </p>
                    </div>
                  </div>
                </div>
              </CityCard>
            </div>
            <div
              className="card-column col-xl-3 col-lg-6 col-md-12"
              key="https://jooble.org/images/cities/us/10460.jpg"
            >
              <CityCard
                city={
                  Object {
                    "cityName": "Atlanta, GA",
                    "imgLink": "https://jooble.org/images/cities/us/10460.jpg",
                    "numOfVacancies": 38.497,
                  }
                }
              >
                <div
                  className="bg-dark text-white city-card card mb-3"
                >
                  <img
                    alt="Card poster"
                    className="card-img"
                    src="https://jooble.org/images/cities/us/10460.jpg"
                  />
                  <div
                    className="card-img-overlay"
                  >
                    <div
                      className="container-fluid"
                      style={
                        Object {
                          "bottom": " 0px",
                          "position": "absolute",
                        }
                      }
                    >
                      <div
                        className="card-title h5"
                      >
                        Atlanta, GA
                      </div>
                      <p
                        className="card-text"
                      >
                        38.497
                      </p>
                    </div>
                  </div>
                </div>
              </CityCard>
            </div>
          </div>
        </div>
      </div>
    </Home>
  </Router>
</StaticRouter>
`;
